### DISCUSSION TITLE
Simple test installation of Mozilla Hubs server.

### DISCUSSION THREAD

**Discussion by salahzar** [mozilla/hubs#3499] [Hubs Cloud Questions] [answered]
I wonder if there are any simple guides to make some experimental builds of Mozilla Hubs server on a private server and local network so that is can be accessible only locally (for experimental purposes or to have it only privately accessible in schools) for test and to understand how it works. So far I have seen that the only apparent way of installing it is on Amazon or Digital Torrent and also need some complex DNS and multiple domains handling and setting up. I tried to set up on kind of localhost, but the instructions got quite difficult for me to follow.
maybe there are some docker images already ready to work to be used?

**Comment by robinkwilson** [comment]
## Question was clarified below, new answer in another comment.

No docker images unfortunately. Here's a few discussions on self-hosting hubs (search for `self hosting` on discussions to pull these up):

Relevant discussion threads:
https://github.com/mozilla/hubs/discussions/3099
https://github.com/mozilla/hubs/discussions/2699
https://github.com/mozilla/hubs/discussions/3049

People *have* set this up themselves, it just takes a lot of work to get everything talking to eachother. Please read the above linked discussions.

The three main systems you would need are (found at the bottom of https://github.com/mozilla/hubs-cloud readme.md): 
https://github.com/mozilla/hubs
https://github.com/mozilla/reticulum - currently, I haven't heard of anyone setting this up locally.
https://github.com/mozilla/dialog

The lightest weight local setup is cloning the `hubs` repo with the `npm run dev` command, you're connecting to our developer backend reticulum + dialog servers.

For running everything locally: Currently there's no way to modify the reticulum backend on a Hubs Cloud deployment and I have not heard of a successful setup for a local version of reticulum running. You'd need to run dialog as well + configure a secret that both dialog + reticulum shares. 

Hopefully this sheds some light on that project. Let me know if you have any more questions.

**Reply by robinkwilson** [reply]
Marking as answer to remove thread from our pipeline, feel free to "Unmark as answer".

**Reply by salahzar** [reply]
Cant mark it as "solved" since i still dont know how to launch even a fraction of it in a reasonable simple way.
If the answer is "you cant unless you are a VERY experienced node programmer and even in that case you will need a lot of time" then  I will close  easily this ticket. 

Just to explain why I asked that, I wanted to use Mozilla Hubs in schools and trying to get a minimum of interaction (clicking on an object making something happens, like popping up a kind of form with a question and an action can be done like teleporting to next question or giving a clue to find next question). This would allow to use Mozilla Hubs as a kind of escape room or scavenger hunt. With standard Mozilla Hubs this appears impossible, while it should be relatively easy to implement this with underlining libraries.

Since you wrote the following:
"The lightest weight local setup is cloning the hubs repo with the npm run dev command, you're connecting to our developer backend reticulum + dialog servers."
I tried it (the two steps) but nothing did happen: when connecting to https://hubs.local:8080/ as reported by npm run dev stdout nothing happened and only a "This site can’t be reached". Surely I need to configure something and or add additional more sophisticated steps, more than just git clone, npm install + npm run.
Is there a simple guide to do that? I tried to read all your links but they are very generic and just stating the difficulty to do that.
I was expecting that if a reasonable path can be done it can be documented in a clear link. Mozilla Hubs is so a good thing that it is a pity that it is so difficult to try on a local installation (at least in part).

**Comment by robinkwilson** [comment, marked as answer]
**Ah thank you for the clarification! Here's the solution to your updated question:**

The AWS Hubs Cloud setup **IS** your own private infrastructure. We designed Hubs Cloud specifically so you have control over everything. Mozilla simply provides you automatic updates to the Hubs Cloud software. The server hardware that AWS provides is under your control. 

More information for locking down your instance is here, especially if you're working with schools: 
[Limiting User Access to your Hub](https://hubs.mozilla.com/docs/hubs-cloud-limiting-user-access.html)

**To get your local test environment setup so you can create interactable components:**
Go to `https://localhost:8080` <- REMEMBER ALWAYS `https`
OR (if there's still issues) follow this guide to setup the hostname https://hubs.local by updating your hosts file. Instructions in the reticulum repository: https://github.com/mozilla/reticulum#1-setup-the-hubslocal-hostname

Then if you're ready to deploy what you've made, you'll need to [setup Hubs Cloud](https://hubs.mozilla.com/cloud) and deploy a custom client: [Deploy a Custom Client](https://hubs.mozilla.com/docs/hubs-cloud-custom-clients.html)

Does this answer your question?

**Reply by robinkwilson** [reply]
Marking as answered, please "unmark" and add more information if we need to revisit.

**Reply by salahzar** [reply]
ok thanks I dont see how I can accept the answer, I'm assuming it has already being marked as answered. Thanks for the attention :)
